pgmpy/models/FactorGraph.py

```diff
@@ -134,11 +137,69 @@ def remove_factors(self, *factors):
         >>> G = FactorGraph()
         >>> G.add_nodes_from(['a', 'b', 'c'])
         >>> phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> G.add_edge('a',phi1)
         >>> G.add_factors(phi1)
         >>> G.remove_factors(phi1)
         """
         for factor in factors:
             self.factors.remove(factor)
+            if factor in super(FactorGraph, self).nodes():
+                super(FactorGraph, self).remove_node(factor)
+
+    def remove_node(self, node):
+        """
+        Removes a single  given node from the added factors.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.models import FactorGraph
+        >>> from pgmpy.factors.discrete import DiscreteFactor
+        >>> G = FactorGraph()
+        >>> G.add_nodes_from(['a', 'b', 'c'])
+        >>> phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> G.add_factors(phi1)
+        >>> G.add_edges_from([('a',phi1),('b',phi1)])
+        >>> G.remove_node('a')
+        """
+        if not isinstance(node, six.string_types):
+            raise ValueError("Node must be a string given{s}", s=type(node))
+
+        affected_factors = [v for u, v in self.edges() if u == node]
+        affected_factors.extend([u for u, v in self.edges() if v == node])
+        # remove and marginalise
+        for factor in affected_factors:
+            edge_nodes = [v for u, v in self.edges() if u == factor]
+            edge_nodes.extend([u for u, v in self.edges() if v == factor])
+            fac_temp = factor.marginalize([node], inplace=False)
+            self.remove_factors(factor)
+            if len(fac_temp.scope()) != 0:
+                self.add_factors(fac_temp)
+                edges = list(zip(edge_nodes, [fac_temp for _ in range(len(edge_nodes))]))
+                self.add_edges_from(edges)
+        super(FactorGraph, self).remove_node(node)
+
+    def remove_nodes_from(self, nodes):
+        """
+        Removes multiple given nodes from the added factors.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.models import FactorGraph
+        >>> from pgmpy.factors.discrete import DiscreteFactor
+        >>> G = FactorGraph()
+        >>> G.add_nodes_from(['a', 'b', 'c'])
+        >>> phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> G.add_factors(phi1)
+        >>> G.add_edges_from([('a',phi1),('b',phi1)])
+        >>> G.remove_nodes_from(['a','b'])
+        """
+        if not isinstance(nodes, (list, tuple, np.ndarray)):
+            raise ValueError("nodes must be a iterable type {s} given".format(s=type(nodes)))
```

Comment:
@kris-singh N should in be caps in `nodes`. Don't use variable name as `s`. Use something that tells what the variable is.